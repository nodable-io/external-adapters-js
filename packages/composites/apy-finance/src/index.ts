import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'APY-Finance'
const { server } = expose(makeExecute())
export { NAME, makeConfig, makeExecute, server }
