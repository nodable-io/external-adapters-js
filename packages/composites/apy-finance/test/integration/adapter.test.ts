import { AdapterRequest, Execute } from '@chainlink/types'
import * as apyFinanceAdapter from "../../src/index"
import { ethers, BigNumber } from 'ethers'

const mockData = {
    '1': {
        symbol: 'WETH',
        balance: 100,
        decimals: 18
    },
    '2': {
        symbol: 'LINK',
        balance: 300,
        decimals: 18
    }
}

const mockRegistry = {
    chainlinkRegistryAddress: () => "mock-address",
    getAssetAllocationIds: () => ['1', '2'],
    symbolOf: (id: string) => mockData[id].symbol,
    balanceOf: (id: string) => mockData[id].balance,
    decimalsOf: (id: string) => mockData[id].decimals 
}

jest.mock("ethers", () => ({
  ...(jest.requireActual("ethers")),
  ethers: {
    providers: {
      JsonRpcProvider: function(_: string): ethers.provider.JsonRpcProvider { return {} } 
    },
    Contract: function() {
      return mockRegistry
    }
  }
}))

jest.mock("@chainlink/token-allocation-adapter", () => ({
  ...(jest.requireActual("@chainlink/token-allocation-adapter")),
  makeExecute: jest.fn().mockReturnValue(() => ({
    "jobRunID": "1",
    "data": {
      "sources": [],
      "payload": {
        "WETH": {
          "quote": {
            "USD": {
              "price": "1800"
            }
          }
        },
        "LINK": {
            "quote": {
              "USD": {
                "price": "2000"
              }
            }
        }
      },
      "result": 2000
    },
    "result": 2000,
    "statusCode": 200
  }))
}))

let oldEnv: NodeJS.ProcessEnv

beforeAll(() => {
  oldEnv = JSON.parse(JSON.stringify(process.env))
  process.env.REGISTRY_ADDRESS = process.env.REGISTRY_ADDRESS || "fake-registry-address" 
})

afterAll(() => {
  process.env = oldEnv
})

describe('execute', () => {
  let execute: Execute 
  const id = '1'

  beforeAll(async () => {
    execute = await apyFinanceAdapter.makeExecute()
  })

  describe("with request properly formatted", () => {
    const data: AdapterRequest = { 
      id,
      data: {
        "source": "tiingo"
      } 
    }

    it("should return success", async () => {
      const resp = await execute(data)
      expect(resp).toMatchSnapshot()
    })
  })
})
