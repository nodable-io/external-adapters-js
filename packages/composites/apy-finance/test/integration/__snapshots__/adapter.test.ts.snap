// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute with request properly formatted should return success 1`] = `
Object {
  "data": Object {
    "payload": Object {
      "LINK": Object {
        "quote": Object {
          "USD": Object {
            "price": "2000",
          },
        },
      },
      "WETH": Object {
        "quote": Object {
          "USD": Object {
            "price": "1800",
          },
        },
      },
    },
    "result": 2000,
    "sources": Array [],
  },
  "jobRunID": "1",
  "result": 2000,
  "statusCode": 200,
}
`;
